src/com/lftechnology/java/training/srijan/basics/coreapp/dao/EmployeeDaoImpl.java

```diff
@@ -0,0 +1,305 @@
+			LOGGER.log(Level.INFO, "Exception:{0}", ex);
+		} finally {
+			try {
+				resultSet.close();
+				prepareStatementUpdateInfo.close();
+				prepareUpdate.close();
+				con.close();
+			} catch (SQLException ex) {
+				LOGGER.log(Level.INFO, "SQL Exception {0}", ex);
+			}
+		}
+	}
+
+	public void updateEmployee(Employee e) {
+
+	}
+
+	public Employee findUser(String userName) {
+		Connection con = null;
+		ResultSet resultSet = null;
+		con = DbConnection.dbConnection();
+		PreparedStatement prepareStatementFindUser = null;
+		Employee user = new Employee();
+		try {
+			if (con != null) {
+				String query = "Select * from employeeDetails where username = ?";
+				prepareStatementFindUser = con.prepareStatement(query);
+				prepareStatementFindUser.setString(1, userName);
+				resultSet = prepareStatementFindUser.executeQuery();
+				while (resultSet.next()) {
+					user.setId(resultSet.getInt("id"));
+					user.setUserName(resultSet.getString("username"));
+					user.setFullName(resultSet.getString("fullName"));
+					user.setDepartment(resultSet.getString("department"));
+					user.setAddress(resultSet.getString("address"));
+					user.setIsTerminated(resultSet.getBoolean("is_terminated"));
+				}
+			}
+		} catch (Exception ex) {
+			LOGGER.log(Level.INFO, "Exception:{0}", ex);
+		} finally {
+			try {
+				resultSet.close();
+				prepareStatementFindUser.close();
+				con.close();
+			} catch (SQLException ex) {
+				LOGGER.log(Level.INFO, "SQL Exception {0}", ex);
+			}
+		}
+		return user;
+	}
+
+	public void updateProfile(Employee changedProfileDetails,Employee employee){
+		Connection con = null;
+		ResultSet resultSet = null;
+		con = DbConnection.dbConnection();
+		PreparedStatement prepareUpdateProfile = null;
+		PreparedStatement prepareUpdate = null;
+		try{
+			if(con!=null){
+				String query = "Select * from employeeDetails where id = ?";
+				prepareUpdateProfile = con.prepareStatement(query);
+				prepareUpdateProfile.setInt(1, employee.getId());
+				resultSet = prepareUpdateProfile.executeQuery();
+				while(resultSet.next()){
+					String queryUpdate = "UPDATE employeeDetails SET username = ?,fullname = ?,department = ?,address = ? where id = ? ";
+					prepareUpdateProfile = con.prepareStatement(queryUpdate);
+					if(!"".equals(changedProfileDetails.getUserName()))
+						prepareUpdateProfile.setString(1, changedProfileDetails.getUserName());
+					else
+						prepareUpdateProfile.setString(1, employee.getUserName());
+					if(!"".equals(changedProfileDetails.getFullName()))
+						prepareUpdateProfile.setString(2, changedProfileDetails.getFullName());
+					else
+						prepareUpdateProfile.setString(2, employee.getFullName());
+					if(!"".equals(changedProfileDetails.getDepartment()))
+						prepareUpdateProfile.setString(3, changedProfileDetails.getDepartment());
+					else
+						prepareUpdateProfile.setString(3, employee.getDepartment());
+					if(!"".equals(changedProfileDetails.getAddress()))	
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)